secret.go

```diff
@@ -93,12 +94,23 @@ func (s Secret) OwnershipValue(fallback Ownership) (ownership Ownership) {
 	return
 }
 
-// Filename returns the expected filename of a secret:  The filename metadata overrides the name
-func (s Secret) Filename() string {
+// Filename returns the expected filename of a secret. The filename metadata overrides the name,
+// but it can't be path, so keysync can't delete or write arbitrary files outside its secrets directory.
+func (s Secret) Filename() (string, error) {
+	name := s.Name
 	if s.FilenameOverride != nil {
-		return *s.FilenameOverride
+		name = *s.FilenameOverride
 	}
-	return s.Name
+
+	if strings.ContainsRune(name, filepath.Separator) {
+		override := ""
+		if s.FilenameOverride != nil {
+			override = *s.FilenameOverride
+		}
+		return "", fmt.Errorf("secret '%s' (FilenameOverride=%s) contains filepath", s.Name, override)
```

Comment:
nit, "contains %s", filepath.Seperator would be easier to understand

Reply:
fair point, I thought maybe it would be beneficial to log actual file path that caused the problem (for debugging purposes)

Reply:
I'd say log both--`"secret '%s' (FilenameOverride=%s) contains filepath separator %s"`--which may've been what @mcpherrinm was suggesting.  That way we know both the problematic path and what was problematic about it.

Reply:
ah yeah I just meant instead of "contains filepath", "contains %s" with the actual seperator